frontend/src/lib/lemon-ui/LemonSlider/LemonSlider.tsx

```diff
@@ -0,0 +1,78 @@
+import clsx from 'clsx'
+import { useEventListener } from 'lib/hooks/useEventListener'
+import { useRef } from 'react'
+
+export interface LemonSliderProps {
+    value?: number
+    onChange?: (value: number) => void
+    min: number
+    max: number
+    /** @default 1 */
+    step?: number
+    className?: string
+}
+
+export function LemonSlider({ value = 0, onChange, min, max, step = 1, className }: LemonSliderProps): JSX.Element {
+    const trackRef = useRef<HTMLDivElement>(null)
+    const movementStartValueWithX = useRef<[number, number] | null>(null)
+
+    useEventListener('mousemove', (e) => {
+        if (e.button === 0 && trackRef.current && movementStartValueWithX.current !== null) {
+            const [movementStartValue, movementStartX] = movementStartValueWithX.current
+            const rect = trackRef.current.getBoundingClientRect()
+            const adjustedWidth = rect.width - 16 // 16px = handle width
+            const deltaX = (e.clientX - movementStartX) / adjustedWidth
+            let newValue = movementStartValue + (max - min) * deltaX
+            newValue = Math.max(min, Math.min(max, newValue)) // Clamped
+            if (step !== undefined) {
+                newValue = Math.round(newValue / step) * step // Adjusted to step
+            }
+            onChange?.(newValue)
+        }
+    })
+    useEventListener('mouseup', (e) => {
+        if (e.button === 0) {
+            movementStartValueWithX.current = null
+        }
+    })
```

Comment:
Flyby: Did you try this on a touch device as well? Usually touch/mouse events are replicated, but sometimes there's some meh involved, such as the background will scroll along (needs preventDefault), or other oddities. Might be totally fine though.

Reply:
That is a great great point, because I forgot about touch. Didn't work. 😅 Now it does. Thanks!